frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -248,6 +248,42 @@ function ReaderHighlight:onTapXPointerSavedHighlight(ges)
     end
 end
 
+function ReaderHighlight:updateHighlight(page, index, side, direction)
+    if self.ui.document.info.has_pages then -- we do this only if it's epub file
+        return
+    end
+
+    local highlight = self.view.highlight.saved[page][index]
+    local highlight_beginning = highlight.pos0
+    local highlight_end = highlight.pos1
+    if side == 0 then -- we move from pos0
+        if direction == 1 then -- move highlight to the right
+            local updated_highlight = self.ui.document:getNextVisibleWordStart(highlight_beginning)
```

Comment:
Vocabulary wise, this is not a `updated_highlight` (a highlight is a pair of xpointers) but a `updated_highlight_beginning`.
I'm still partial to using the common pos0 / pos1 vocabulary instead of _beginning or _end thus.